### PR TITLE
ert hosting test

### LINKED ISSUE #2288: Edge router/tunneler hosting Chaos Test
Add chaos testing for edge router tunneler hosting. Ensure that terminators remain consistent as controllers and routers are restarted.

### PR DESCRIPTION
**Pull request by plorenz**
- **Add ERT hosting chaos test. Also add ert terminator validation utility. Fixes #2288**
- **Rework ER/T terminator management based on SDK terminator management code**

### CODE REVIEW

File: router/config.go [hunk 1]
```diff
@@ -885,12 +885,14 @@ func LoadConfigWithOptions(path string, loadIdentity bool) (*Config, error) {
...
 			if a != nil {
-				// should start with tls:
-				addy := strings.TrimPrefix(a.(string), "tls:")
-				addy = strings.Split(addy, ":")[0]
-				e := cfg.Id.ValidFor(addy)
-				if e != nil {
-					errs = append(errs, fmt.Errorf("invalid link.listeners.advertise: %s, error: %v", a.(string), e))
+				// should start with tls: or dtls:
+				parts := strings.Split(a.(string), ":")
+				if parts[0] == "tls" || parts[0] == "dtls" {
```
Comment (andrewpmartinez) [inline router/config.go:890]: Seems odd that we would be explicitly checking for `tsl` and `dtls`. Are there other types of listeners we would be filtering out? The only one I can think of is the web socket listener on routers, but IIRC that isn't applicable for terminators.

File: router/xgress_edge_tunnel_v2/dialer.go [hunk 2]
```diff
@@ -82,3 +82,10 @@ func (self *tunneler) Dial(params xgress.DialParams) (xt.PeerData, error) {
 
 	return peerData, nil
 }
+
+func (self *tunneler) Inspect(key string, timeout time.Duration) any {
+	if key == "ert-terminators" {
```
Comment (andrewpmartinez) [inline router/xgress_edge_tunnel_v2/dialer.go:87]: I've seen this value a few times. Is there a reason it isn't a constant? It looks meaningful.

File: router/xgress_edge_tunnel_v2/tunneler.go [hunk 3]
```diff
@@ -120,33 +115,31 @@ func (self *tunneler) Start(notifyClose <-chan struct{}) error {
...
-					self.terminators.Remove(key)
-					pfxlog.Logger().Debugf("removed closed tunnel terminator %v for service %v", key, t.context.ServiceName())
-				}
-			}
-		case <-notifyClose:
-			return
-		}
-	}
-}
+//func (self *tunneler) removeStaleConnections(notifyClose <-chan struct{}) {
```
Comment (andrewpmartinez) [inline router/xgress_edge_tunnel_v2/tunneler.go:118]: dead?

File: zititest/models/ert-hosting-test/main.go [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+					"protocol" : "tcp"
+				}`))
+
+			workflow.AddAction(model.ActionFunc(func(run model.Run) error {
+				//encryptionRequired := true
+				var tasks []parallel.LabeledTask
+				for i := 0; i < 2000; i++ {
+					name := fmt.Sprintf("service-%04d", i)
+					//
+					//task := parallel.TaskWithLabel("create.service", "create new service", func() error {
```
Comment (andrewpmartinez) [inline zititest/models/ert-hosting-test/main.go:266]: dead?

**Review by andrewpmartinez — APPROVED** [review]

File: router/config.go (hunk 1 above)
Comment (plorenz) [reply router/config.go:890]: Clint had changed this to only allow tls, I'm broadening it to dtls. I think we can tweak it if/when we add another protocol. Might be able to check it against transport.ParseAddress.

File: router/xgress_edge_tunnel_v2/dialer.go (hunk 2 above)
Comment (plorenz) [reply router/xgress_edge_tunnel_v2/dialer.go:87]: fixed

File: router/xgress_edge_tunnel_v2/tunneler.go (hunk 3 above)
Comment (plorenz) [reply router/xgress_edge_tunnel_v2/tunneler.go:118]: fixed in follow-up PR
